lib/action_subscriber/publisher/async/in_memory_adapter.rb

```diff
@@ -62,6 +62,7 @@ class AsyncQueue
 
           def initialize
             @queue = ::Queue.new
+            @instrumentation_threadpool = ::Lifeguard::Threadpool.new
```

Comment:
I could make this a single thread because Lifeguard won't block, but I left this as default for now.